### PR TITLE
Make HTTPHeaderValue Codable

### PR DESCRIPTION
**Pull request by MrLotU**

### CODE REVIEW

File: Sources/HTTPKit/Utilities/HTTPHeaderValue.swift [hunk 1]
```diff
@@ -197,3 +197,26 @@ public struct HTTPHeaderValue {
         return HTTPHeaderValue(.init(value), parameters: parameters)
     }
 }
+
+extension HTTPHeaderValue: Codable {
```
Comment (tanner0101) [inline Sources/HTTPKit/Utilities/HTTPHeaderValue.swift]: oh gosh, now GitHub thinks the rest of the file is a comment xD

File: Sources/HTTPKit/Utilities/HTTPHeaderValue.swift [hunk 2]
```diff
@@ -197,3 +197,26 @@ public struct HTTPHeaderValue {
...
+        let container = try decoder.singleValueContainer()
+        let string = try container.decode(String.self)
+        guard let tempValue = HTTPHeaderValue.parse(string) else {
+            throw DecodingError.dataCorruptedError(in: container, debugDescription: "Invalid header value string")
+        }
+        self.parameters = tempValue.parameters
+        self.value = tempValue.value
+    }
+    
+    /// Encode a HTTPHeaderValue into an Encoder
```
Comment (tanner0101) [inline Sources/HTTPKit/Utilities/HTTPHeaderValue.swift]: type names in doc blocks should be back-ticked to make them clickable in generated docs

File: Sources/HTTPKit/Utilities/HTTPHeaderValue.swift (hunk 1 above)
Comment (MrLotU) [reply Sources/HTTPKit/Utilities/HTTPHeaderValue.swift]: Yup, haha, they have a lot to learn about the new raw strings

File: Sources/HTTPKit/Utilities/HTTPHeaderValue.swift (hunk 2 above)
Comment (MrLotU) [reply Sources/HTTPKit/Utilities/HTTPHeaderValue.swift]: Will fix!

**Comment by penny-coin** [comment]
Hey @MrLotU, you just merged a pull request, have a coin! 

You now have 77 coins.

**Pull request merged**